pkg/appStore/deployment/AppStoreDeploymentNewService.go

```diff
@@ -0,0 +1,498 @@
+			InstalledAppVersionId:        a.InstalledAppVersionId,
+			InstalledAppsId:              a.InstalledAppId,
+			EnvironmentId:                a.EnvironmentId,
+			AppOfferingMode:              a.AppOfferingMode,
+		}
+
+		// if hyperion mode app, then fill clusterId and namespace
+		if a.AppOfferingMode == util2.SERVER_MODE_HYPERION {
+			environment, err := impl.environmentRepository.FindById(a.EnvironmentId)
+			if err != nil {
+				impl.logger.Errorw("fetching environment error", "err", err)
+				return nil, err
+			}
+			installedAppRes.ClusterId = environment.ClusterId
+			installedAppRes.Namespace = environment.Namespace
+		}
+
+		installedAppsEnvResponse = append(installedAppsEnvResponse, installedAppRes)
+	}
+	return installedAppsEnvResponse, nil
+}
+
+func (impl AppStoreDeploymentServiceImpl) DeleteInstalledApp(ctx context.Context, installAppVersionRequest *appStoreBean.InstallAppVersionDTO) (*appStoreBean.InstallAppVersionDTO, error) {
+
+	environment, err := impl.environmentRepository.FindById(installAppVersionRequest.EnvironmentId)
+	if err != nil {
+		impl.logger.Errorw("fetching error", "err", err)
+		return nil, err
+	}
+
+	dbConnection := impl.installedAppRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		return nil, err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+
+	app, err := impl.appRepository.FindById(installAppVersionRequest.AppId)
+	if err != nil {
+		return nil, err
+	}
+	app.Active = false
+	app.UpdatedBy = installAppVersionRequest.UserId
+	app.UpdatedOn = time.Now()
+	err = impl.appRepository.UpdateWithTxn(app, tx)
+	if err != nil {
+		impl.logger.Errorw("error in update entity ", "entity", app)
+		return nil, err
+	}
+
+	model, err := impl.installedAppRepository.GetInstalledApp(installAppVersionRequest.InstalledAppId)
+	if err != nil {
+		impl.logger.Errorw("error in fetching installed app", "id", installAppVersionRequest.InstalledAppId, "err", err)
+		return nil, err
+	}
+	model.Active = false
+	model.UpdatedBy = installAppVersionRequest.UserId
+	model.UpdatedOn = time.Now()
+	_, err = impl.installedAppRepository.UpdateInstalledApp(model, tx)
+	if err != nil {
+		impl.logger.Errorw("error while creating install app", "error", err)
+		return nil, err
+	}
+	models, err := impl.installedAppRepository.GetInstalledAppVersionByInstalledAppId(installAppVersionRequest.InstalledAppId)
+	if err != nil {
+		impl.logger.Errorw("error while fetching install app versions", "error", err)
+		return nil, err
+	}
+	for _, item := range models {
+		item.Active = false
+		item.UpdatedBy = installAppVersionRequest.UserId
+		item.UpdatedOn = time.Now()
+		_, err = impl.installedAppRepository.UpdateInstalledAppVersion(item, tx)
+		if err != nil {
+			impl.logger.Errorw("error while fetching from db", "error", err)
+			return nil, err
+		}
+	}
+
+	if util2.GetDevtronVersion().ServerMode == util2.SERVER_MODE_HYPERION || app.AppOfferingMode == util2.SERVER_MODE_HYPERION  {
```

Comment:
we can apply RBAC in the earlier

Reply:
This is Not RBAC. This is business logic